### ISSUE TITLE
GitHub Pages fails if nothing to commit again

### ISSUE THREAD

**Issue by djhoese** [travis-ci/dpl#1216]
See #1095 for the original issue and #1097 for the original fix. It seems this is an issue again in dpl2. I don't know ruby so I had trouble figuring out what was going on in https://github.com/travis-ci/dpl/blob/master/lib/dpl/providers/pages/git.rb, but my deployments are failing when there is nothing to commit.

```
Cloning the branch master from the remote repo
Copying /home/travis/build/vispy/vispy/doc/_build/html contents to /tmp/d20200922-17894-1tqiu6d
$ rsync -rl --exclude .git --delete "/home/travis/build/vispy/vispy/doc/_build/html/" .
Preparing to deploy master branch to gh-pages
$ echo "vispy.org" > CNAME
$ git add -A .
$ git commit -q -m "Deploy vispy.org website for SHA:81e248df5d0cb492e8c8fe15052dfc469898b4d2 (Tag: )"
On branch master
Your branch is up to date with 'origin/master'.
nothing to commit, working tree clean

Failed
failed to deploy
```

https://travis-ci.com/github/vispy/vispy/jobs/388864952

**Comment by suhrmann** [comment]
Same problem here in project [Anno-1800-Calculator](https://github.com/suhrmann/Anno-1800-Calculator)

Sample: [failed task on Windows using ``electron-builder``, Travis-CI](https://travis-ci.org/github/suhrmann/Anno-1800-Calculator/jobs/740636161)

Environment:
```
.travis.yml:
  - os: windows
    language: node_js
    node_js: "lts/*"

package.json:
    "electron": "^9.0.0",
    "electron-devtools-installer": "^3.1.0",
```

**Comment by andrewdavidmackenzie** [comment]
Close as a duplicate of https://github.com/travis-ci/dpl/issues/1095 ??

**Comment by djhoese** [comment]
Maybe, but according to that issue it was fixed in dpl version 1, but my error is still occurring in dpl version 2. It is possible that the fix never got ported or that the fix no longer works, but I was still seeing this issue a year after #1097 was merged (I think). If it wasn't merged/ported to dpl2 or is not currently working still then this is not a pure duplicate.

**Comment by andrewdavidmackenzie** [comment]
Agreed the issue is not fixed (fails for me), but better to have only one issue for that, no?

**Comment by djhoese** [comment]
Sure but then that issue needs to be reopened. That issue was also for dpl v1 so maybe a second issue is OK? Since the issue was fixed and is now not fixed I assume it is a different cause. I guess it is maintainer preference on how this is treated. :shrug:

**Comment by andrewdavidmackenzie** [comment]
OK, makes sense. I'll follow both.
Is there a way to select the version of deploy do you know?

**Comment by djhoese** [comment]
Nope, sorry. Only this: https://github.com/travis-ci/dpl/issues/1095#issuecomment-530448622

**Comment by andrewdavidmackenzie** [comment]
Yeh, I tried that - and didn't see any difference in behaviour...
